### PR TITLE
Add support for the CSP nonce

### LINKED ISSUE #55: Content Security Policy
@markets 👋 

I have enabled a [Content Security Policy](https://edgeguides.rubyonrails.org/security.html#content-security-policy) but that breaks invisible captcha since inlined CSS changes every second. I can make it work by setting `style-src` to `unsafe-inline` but that's more of a hack.

Is there a way to insert a `nonce` to a generated`<style>` tag?

### PR DESCRIPTION
**Pull request by jmstfv**
👋 @markets 

I have replaced `v0.12.2` with this branch in my Gemfile and it worked without `unsafe-inline` 🎉 

The question is: should `nonce` be passed as a [standalone attribute](https://github.com/markets/invisible_captcha/compare/master...jmstfv:csp?expand=1#diff-9a7e86091679643c8aa1386d20eab83fR61) or should we pass the entire `options` hash?

Closes #55

### CODE REVIEW

**Comment by markets** [comment]
Hi @jmstfv 

I think passing the hash is fine, since it's already used for other options as well.

**Comment by jmstfv** [comment]
@markets No idea why that test failed. I think I will revert to the previous working [commit](https://github.com/markets/invisible_captcha/pull/61/commits/5ffff3c34508d28e1d33836511f1be03073dbb9a)

**Comment by markets** [comment]
Yep, no problem, tbh after a second pass :eyes: the first commit seems totally fine :+1: 

We'll need to update the docs as well.

**Comment by jmstfv** [comment]
I wrote the following test in `view_helpers_spec.rb` but it failed.

```
it 'with CSP nonce' do
  expect(invisible_captcha(:subtitle, :topic, { nonce: true })).to match(/nonce=/)
end
```

```
Failures:

  1) InvisibleCaptcha::ViewHelpers with custom html options
     Failure/Error: nonce = content_security_policy_nonce if options[:nonce] == true

     NameError:
       undefined local variable or method `content_security_policy_nonce' for #<RSpec::ExampleGroups::InvisibleCaptchaViewHelpers:0x00007fa3b4321d10>
     # ./lib/invisible_captcha/view_helpers.rb:59:in `visibility_css'
     # ./lib/invisible_captcha/view_helpers.rb:37:in `build_invisible_captcha'
     # ./lib/invisible_captcha/view_helpers.rb:16:in `invisible_captcha'
     # ./spec/view_helpers_spec.rb:32:in `block (2 levels) in <top (required)>'
```
Seems like RSpec can't find [content_security_policy_nonce](https://edgeapi.rubyonrails.org/classes/ActionDispatch/ContentSecurityPolicy/Request.html#method-i-content_security_policy_nonce). Do you know how to include that?

**Comment by markets** [comment]
hello @jmstfv to be honest I have no idea (never implemented this new CSP yet). This spec file is marked as "helper" test:

```
RSpec.describe InvisibleCaptcha::ViewHelpers, type: :helper do
```

I supose `rspec-rails` will need to recognize the `content_security_policy_nonce` automatically, but it doesn't seem so... maybe we should ask Rails or Rspec teams how to test this thing?

**Comment by jmstfv** [comment]
But it was working on my machine... 😃

This test only works on Rails 5.2 and up.

**Comment by jmstfv** [comment]
Yikes, forgot to add the if statement for the test.

**Comment by jmstfv** [comment]
Closes #55

File: lib/invisible_captcha/view_helpers.rb [hunk 1]
```diff
@@ -56,7 +56,13 @@ def visibility_css(css_class, options)
 
       return if visible
 
-      content_tag(:style, media: 'screen') do
+      nonce = if ::Rails::VERSION::STRING >= '5.2'
```
Comment (markets) [inline lib/invisible_captcha/view_helpers.rb]: maybe we can use the simpler (and already used in other places) statement `Rails.version`? I'd also replace the other occurrences.

Comment (jmstfv) [reply lib/invisible_captcha/view_helpers.rb]: Sounds good to me! Replaced them all with this snippet: 

`find . -name '*.rb' -print0 | xargs -0 sed -i "" "s/::Rails::VERSION::STRING/Rails.version/g`

**Comment by jmstfv** [comment]
Don't forget to tag this release as `v0.13.0` and push it to RubyGems

**Comment by markets** [comment]
> Don't forget to tag this release as `v0.13.0` and push it to RubyGems

I'll do it this afternoon/night!

**Comment by markets** [comment]
@jmstfv `v0.13.0` pushed to RubyGems: https://rubygems.org/gems/invisible_captcha/versions/0.13.0 :rocket: 

Thanks a lot for your work! :clap:

**Comment by jmstfv** [comment]
Thank you too, @markets 

I will push it to production tomorrow and see how it goes.

**Comment by birthdaycorp** [comment]
Sorry to drag up an old issue, but I'm noticing in my app that adding `nonce: true` to the helper adds the nonce to `script-src`. This ends up breaking my site. Shouldn't it go to `style-src`?

**Comment by jmstfv** [comment]
@birthdaycorp you can confine the nonce attribute to `style-src` directive inside `config/initializers/content_security_policy.rb`:

```
# Set the nonce only to specific directives
Rails.application.config.content_security_policy_nonce_directives = %w(style-src)
```

**Comment by birthdaycorp** [comment]
> Sorry to drag up an old issue, but I just noticed that adding `nonce: true` to the `<%= invisible_captcha %>` helper adds the nonce to `script-src` in the header. This ends up breaking my site. Since the nonce is injected into the style tag, shouldn't it go to `style-src`?

Hmm, it's still added to `script-src` only. But I need nonces for both, so I can't do that. Do you have any idea how I might work around this?

**Comment by jmstfv** [comment]
@birthdaycorp if you want nonce added to both `script-src` and `style-src` directives:

```
Rails.application.config.content_security_policy_nonce_directives = %w(script-src style-src)
```

Have you restarted the server after making this change?

```
rails restart
```

**Comment by birthdaycorp** [comment]
@jmstfv yeah i've tried both. i'm following the docs, but the nonce is never added to style-src.

**Pull request merged**